<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>RuntimeSetting</class>
 <widget class="QWidget" name="RuntimeSetting">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>782</width>
    <height>524</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>782</width>
    <height>524</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>系统运行设置</string>
  </property>
  <property name="windowIcon">
   <iconset resource="MainWIndow.qrc">
    <normaloff>:/MainWIndow/tool_images/config_setting.png</normaloff>:/MainWIndow/tool_images/config_setting.png</iconset>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <widget class="QGroupBox" name="groupBox_7">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>初始化</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>图片处理</string>
        </property>
        <layout class="QFormLayout" name="formLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>最大特征数</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="max_features">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximum">
            <number>999999999</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>SIFT Contrast
Threshold</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="sift_contrast_threshold">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="decimals">
            <number>10</number>
           </property>
           <property name="minimum">
            <double>-9999999999.000000000000000</double>
           </property>
           <property name="maximum">
            <double>9999999999.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>SIFT Sigma</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QDoubleSpinBox" name="sitf_sigma">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="decimals">
            <number>10</number>
           </property>
           <property name="minimum">
            <double>-9999999999.000000000000000</double>
           </property>
           <property name="maximum">
            <double>9999999999.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>最近/次近距离比</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QDoubleSpinBox" name="dist_ratio_in_filter_matches">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="decimals">
            <number>10</number>
           </property>
           <property name="minimum">
            <double>-9999999999.000000000000000</double>
           </property>
           <property name="maximum">
            <double>9999999999.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>图片最小特征匹配数</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QSpinBox" name="min_matches_edge_in_create_image_graph">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximum">
            <number>999999999</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="title">
         <string>地图初始化</string>
        </property>
        <layout class="QFormLayout" name="formLayout_2">
         <item row="0" column="0">
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>2D-2D
最小内点数</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="min_inliers_in_2d2d_matching">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximum">
            <number>999999999</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>H E RANSAC 置信度</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="ransac_confidence_in_compute_H_E">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="decimals">
            <number>10</number>
           </property>
           <property name="minimum">
            <double>-9999999999.000000000000000</double>
           </property>
           <property name="maximum">
            <double>9999999999.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_8">
           <property name="text">
            <string>E 误差阈值</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QDoubleSpinBox" name="max_error_in_compute_F_E">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="decimals">
            <number>10</number>
           </property>
           <property name="minimum">
            <double>-9999999999.000000000000000</double>
           </property>
           <property name="maximum">
            <double>9999999999.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_9">
           <property name="text">
            <string>H 误差阈值</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QDoubleSpinBox" name="max_error_in_compute_H">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="decimals">
            <number>10</number>
           </property>
           <property name="minimum">
            <double>-9999999999.000000000000000</double>
           </property>
           <property name="maximum">
            <double>9999999999.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_10">
           <property name="text">
            <string>重投影误差阈值</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QDoubleSpinBox" name="max_error_in_init_triangulte">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="decimals">
            <number>10</number>
           </property>
           <property name="minimum">
            <double>-9999999999.000000000000000</double>
           </property>
           <property name="maximum">
            <double>9999999999.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="label_11">
           <property name="text">
            <string>最小角度</string>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QDoubleSpinBox" name="min_angle_in_init_triangulte">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="decimals">
            <number>10</number>
           </property>
           <property name="minimum">
            <double>-9999999999.000000000000000</double>
           </property>
           <property name="maximum">
            <double>9999999999.000000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="QGroupBox" name="groupBox_6">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>增量式 SfM</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QGroupBox" name="groupBox_3">
        <property name="title">
         <string>位姿求解</string>
        </property>
        <layout class="QFormLayout" name="formLayout_3">
         <item row="0" column="0">
          <widget class="QLabel" name="label_12">
           <property name="text">
            <string>2D-3D 最小匹配数</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="min_inlers_in_pnp">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximum">
            <number>999999999</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_13">
           <property name="text">
            <string>RANSAC 最大迭代数</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="ransac_times_in_pnp">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximum">
            <number>999999999</number>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_14">
           <property name="text">
            <string>RANSAC 置信度</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QDoubleSpinBox" name="ransac_conifidence_in_pnp">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="decimals">
            <number>10</number>
           </property>
           <property name="minimum">
            <double>-9999999999.000000000000000</double>
           </property>
           <property name="maximum">
            <double>9999999999.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_15">
           <property name="text">
            <string>重投影误差阈值</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QDoubleSpinBox" name="max_error_in_pnp">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="decimals">
            <number>10</number>
           </property>
           <property name="minimum">
            <double>-9999999999.000000000000000</double>
           </property>
           <property name="maximum">
            <double>9999999999.000000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_4">
        <property name="title">
         <string>三角测量</string>
        </property>
        <layout class="QFormLayout" name="formLayout_4">
         <item row="0" column="0">
          <widget class="QLabel" name="label_16">
           <property name="text">
            <string>重投影误差阈值</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QDoubleSpinBox" name="max_error_in_triangulate">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="decimals">
            <number>10</number>
           </property>
           <property name="minimum">
            <double>-9999999999.000000000000000</double>
           </property>
           <property name="maximum">
            <double>9999999999.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_17">
           <property name="text">
            <string>最小角度</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="min_anglue_in_triangulate">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="decimals">
            <number>10</number>
           </property>
           <property name="minimum">
            <double>-9999999999.000000000000000</double>
           </property>
           <property name="maximum">
            <double>9999999999.000000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_5">
        <property name="title">
         <string>捆绑优化</string>
        </property>
        <layout class="QFormLayout" name="formLayout_5">
         <item row="0" column="0">
          <widget class="QLabel" name="label_18">
           <property name="text">
            <string>重投影误差阈值</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QDoubleSpinBox" name="max_error_in_BA_filter">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="decimals">
            <number>10</number>
           </property>
           <property name="minimum">
            <double>-9999999999.000000000000000</double>
           </property>
           <property name="maximum">
            <double>9999999999.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_19">
           <property name="text">
            <string>最小角度</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="min_angle_in_BA_filter">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="decimals">
            <number>10</number>
           </property>
           <property name="minimum">
            <double>-9999999999.000000000000000</double>
           </property>
           <property name="maximum">
            <double>9999999999.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_20">
           <property name="text">
            <string>LBA 最大图片数</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QSpinBox" name="max_image_count_in_localBA">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximum">
            <number>999999999</number>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_21">
           <property name="text">
            <string>触发 GBA 的最大图片数</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QSpinBox" name="max_found_image_count_in_BA_to_run_globalBA">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximum">
            <number>999999999</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="1">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="MainWIndow.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>RuntimeSetting</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>613</x>
     <y>496</y>
    </hint>
    <hint type="destinationlabel">
     <x>350</x>
     <y>492</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>RuntimeSetting</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>715</x>
     <y>496</y>
    </hint>
    <hint type="destinationlabel">
     <x>219</x>
     <y>489</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>accept()</slot>
  <slot>reject()</slot>
 </slots>
</ui>
